### PR TITLE
component-implementation

### PR DESCRIPTION
**Pull request by dsusviela**
Gather.js:
  - Added the modal component
  - Added touchableOpacity wrapper arround Mapboxes pointAnnotation for a neat animation on press of the Isles
  - Added redux dependencies for the createPocketModal component

gather/styles.js:
  - Added several new styles for the modal component

CreatePocketModal.js:
  - Changed "in" animation to match gather.js compoenent's animation
  - Changed invalid number description error when a wrong id was passed to a more descriptive one.

Drawer.js:
  - Removed all redux dependencies for the createPocketModal component
  - Removed all history button functionality

localization/en.js and localization/es.js:
  - Added several new strings for the new component

locaclization/es.js:
  - Changed `invalidInputNumber` string to a more descriptive one.

**Please note if yore having problems with getting the Isles to show the component you might need to run the following command: `npm cache clean --force`**

### CODE REVIEW

File: src/components/Gather/Gather.js [hunk 1]
```diff
@@ -24,6 +27,39 @@ import GatherOverlay from './GatherOverlay';
...
 
+const GatherPointOptionModal = ({ isVisible, onPressActionFst, onPressActionSnd }) =>
+  (
+    <Modal
+      isVisible={isVisible}
+      onBackdropPress={onPressActionFst}
+      onBackButtonPress={onPressActionFst}
+      animationOut="slideOutLeft"
+      backdropOpacity="0.3"
+      backdropColor={Colors.primary}
```
[the remark this answers is not in the document]
Comment (dsusviela) [reply src/components/Gather/Gather.js]: 0.7 is the rest of the app. My idea for changing the color was to make it clear to the user that the "trip" was still on course even through the menus. Ill change it.

File: src/components/Gather/Gather.js [hunk 2]
```diff
@@ -24,6 +27,39 @@ import GatherOverlay from './GatherOverlay';
...
+    >
+      <View style={stylesGather.modalContainer}>
+        <View style={stylesGather.modalTitleContainer}>
+          <Text style={stylesGather.modalTitle}>{strings.optionsModalGather}</Text>
+        </View>
+        <View>
+          <Button
```
[the remark this answers is not in the document]
Comment (dsusviela) [reply src/components/Gather/Gather.js]: The only reason being that the story had a certain element of urgency, and i wanted to get it done asap. Ill add the icons with the other fixes, of course

File: src/components/Gather/Gather.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (anitanigma) [reply src/components/Gather/Gather.js]: Ohh I see, I think it's a cool idea, but we'll have the finish "trip" button in the future so I don't think we need to worry about that now

File: src/components/Gather/Gather.js [hunk 3]
```diff
@@ -24,6 +27,39 @@ import GatherOverlay from './GatherOverlay';
...
+            style={stylesGather.buttonModal}
+            textStyle={stylesGather.text}
+            title={strings.changeStateIsle}
```
[the remark this answers is not in the document]
Comment (anitanigma) [reply src/components/Gather/Gather.js]: Sounds good to me !

**Comment by dsusviela** [comment]
Quick update on this;

Colors have been changed, it has been merged with master and ive changed the way icons look in the first modal. 

What ive been unable to solve is the issue with the empty annotations. After hours of googling this looks like theres something going on with mapbox. We can change the implementation of the isles, but i feel like that would exceed the 2 points of this story significantly (which btw its already taking longer). My suggestion is to leave this as is until i get a reply or something from [https://github.com/mapbox/react-native-mapbox-gl/issues/1413](url).

**Pull request merged**